src/lib/cloud-sdk-api.js

```diff
@@ -191,6 +245,141 @@ class CloudSdkAPI {
       throw `Error: ${change.status} - ${change.statusText}`
     }
   }
+
+  async _requestJson(callback) {
+    let response = await callback();
+    if (response.status === 200) {
+      return await (response.json());
+    } else {
+      throw `Error: ${response.status} - ${response.statusText}`
+    }
+  }
+
+  async _waitForStatus(predicate, request) {
+      let status = await this._requestJson(request);
+
+      while (!predicate(status)) {
+        await sleepSeconds(10);
+        status = await this._requestJson(request);
+      }
+
+      return status;
+  }
+
+  async _waitForEnvRunning(namespace) {
+    await this._waitForEnv(namespace, 'running');
+  }
+
+  async _waitForEnvHibernated(namespace) {
+    await this._waitForEnv(namespace, 'hibernated');
+  }
+
+  async _waitForEnvRunningOrHibernated(namespace) {
+    return await this._waitForEnv(namespace, 'running', 'hibernated');
+  }
+
+  async _waitForEnv(namespace, state1, state2) {
+    return (await this._waitForStatus(
+      (status) => status.releases?.status[`cm-p${this._programId}-e${this._environmentId}`]?.releaseState === state1 ||
+                  status.releases?.status[`cm-p${this._programId}-e${this._environmentId}`]?.releaseState === state2,
+      async () => await this._doDevConsoleRequest(`get`,`/api/releases/${namespace}/status`)
+    )).releases.status[`cm-p${this._programId}-e${this._environmentId}`].releaseState;
+  }
+
+  async _hibernateEnv(namespace) {
+    await this._setEnvStatus(namespace, `hibernate`);
+  }
+
+  async _dehibernateEnv(namespace) {
+    await this._setEnvStatus(namespace, `dehibernate`);
+  }
+
+  async _setEnvStatus(namespace, status) {
+    await this._waitForStatus(
+      (status) => status.ok,
```

Comment:
`status` in the Closure covers the argument `status`, which I find confusing. Unless I am confused and misunderstand the code :)

Reply:
yeah, I think we can call it target

Reply:
👍